### ISSUE TITLE
false positive for `:unused-value` linter with transient-mutating functions like `assoc!`

### ISSUE THREAD

**Issue by camsaul** [clj-kondo/clj-kondo#1931]
**version**

```
clj-kondo v2022.12.10
```

**platform**

Native version installed via Homebrew

**problem**

Functions that mutate transients like `assoc!` can result in a `:unused-value` warning. Functions like `assoc!` should probably be exempt from the `:unused-value` linter since they can be done for side effects.

**repro**

```clj
(defn my-assoc! [m k v]
  (assoc! m k v)
  m)
```

**config**

This is reproducible in an empty project with no Kondo config

**expected behavior**

No error in this situation

**Comment by camsaul** [comment]
Ok I guess actually this warning does make sense considering that you're supposed to use the value returned by `assoc!`
